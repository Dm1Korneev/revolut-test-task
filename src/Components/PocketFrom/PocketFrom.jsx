import React from 'react';

import Pocket from 'Components/Pocket';

function PocketFrom(props) {
  return (
    <Pocket
      {...props}
      valuePrefix="-"
      inputValueLabel="Write off amount"
      selectCurrencyLabel="Write off currency"
    />
  );
}

export default PocketFrom;
