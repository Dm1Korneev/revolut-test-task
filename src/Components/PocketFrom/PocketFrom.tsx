import React, { FC } from 'react';

import Pocket, { Props as PocketProps } from 'Components/Pocket';

const PocketFrom: FC<PocketProps> = (props: PocketProps): JSX.Element => (
  <Pocket
    {...props}
    valuePrefix="-"
    inputValueLabel="Write off amount"
    selectCurrencyLabel="Write off currency"
  />
);

export default PocketFrom;
