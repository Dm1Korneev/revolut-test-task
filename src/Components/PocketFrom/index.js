import commonHoc from 'Components/commonHoc';
import { changeWriteOffValue, setPocketFrom } from 'Redux/actions';
import {
  pocketFromCurrencySelector, pocketFromValueSelector, writeOffValueSelector,
} from 'Selectors/exchange';
import { currenciesSelector } from 'Selectors/currencies';

import PocketFrom from './PocketFrom';

const mapStateToProps = (state) => ({
  currency: pocketFromCurrencySelector(state),
  valueHave: pocketFromValueSelector(state),
  currencies: currenciesSelector(),
  value: writeOffValueSelector(state),
});

const mapDispatchToProps = {
  setPocketCurrency: setPocketFrom,
  changeValue: changeWriteOffValue,
};

export default commonHoc(PocketFrom, {
  mapStateToProps,
  mapDispatchToProps,
});
