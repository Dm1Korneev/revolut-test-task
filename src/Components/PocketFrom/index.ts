import commonHoc from 'Components/commonHoc';
import { changeWriteOffValue, setPocketFrom } from 'Redux/actions';
import {
  pocketFromCurrencySelector, pocketFromValueSelector, writeOffValueSelector,
} from 'Selectors/exchange';
import { currenciesSelector } from 'Selectors/currencies';
import { StateType } from 'Redux/store';

import PocketFrom from './PocketFrom';

const mapStateToProps = (state: StateType): Record<string, any> => ({
  currency: pocketFromCurrencySelector(state),
  valueHave: pocketFromValueSelector(state),
  currencies: currenciesSelector(),
  value: writeOffValueSelector(state),
});

const mapDispatchToProps = {
  setPocketCurrency: setPocketFrom,
  changeValue: changeWriteOffValue,
};

export default commonHoc(PocketFrom, {
  mapStateToProps,
  mapDispatchToProps,
});
