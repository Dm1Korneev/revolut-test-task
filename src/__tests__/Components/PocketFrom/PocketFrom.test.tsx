import React from 'react';
import { render } from '@testing-library/react';
import PocketFrom from 'Components/PocketFrom/PocketFrom';

jest.mock('Components/Pocket', () => mockComponent('Pocket'));

describe('PocketFrom component', () => {
  const props = {
    currency: 'USD',
    currencies: ['USD', 'EUR', 'GBP'],
    setPocketCurrency: (): void => {},
    changeValue: (): void => {},
    inputValueLabel: 'inputValueLabelTest',
    selectCurrencyLabel: 'selectCurrencyLabeltsTest',
  };

  test('renders the component', () => {
    const { asFragment } = render(<PocketFrom {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
