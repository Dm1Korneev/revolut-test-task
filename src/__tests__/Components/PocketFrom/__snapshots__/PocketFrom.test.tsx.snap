// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PocketFrom component renders the component 1`] = `
<DocumentFragment>
  <div
    data-originalcomponent="Pocket"
    test-prop-currencies="[\\"USD\\",\\"EUR\\",\\"GBP\\"]"
    test-prop-currency="\\"USD\\""
    test-prop-inputvaluelabel="\\"Write off amount\\""
    test-prop-selectcurrencylabel="\\"Write off currency\\""
    test-prop-valueprefix="\\"-\\""
  />
</DocumentFragment>
`;
